### PR TITLE
MAINT: set same tolerance between LSMR and LSQR

### LINKED ISSUE #13745: different default atol, btol for lsqr, lsmr
The default `atol` and `btol` for [lsqr](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lsqr.html) is `1.0e-8`, but only `1.0e-6` for [lsmr](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lsmr.html).

### PR DESCRIPTION
**Pull request by tupui**
Closes #13745

Setting the same tolerance between the two algorithm would allow practitioner to swap from one algorithm and have the same defaults.

### CODE REVIEW

File: scipy/sparse/linalg/isolve/lsmr.py [hunk 1]
```diff
@@ -141,6 +141,7 @@ def lsmr(A, b, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8,
 
     Examples
     --------
+    >>> import numpy as np
```
Comment (rgommers) [inline scipy/sparse/linalg/isolve/lsmr.py:144]: this is unnecessary, and we try to avoid adding it to every single example; the import is assumed.

Comment (tupui) [reply scipy/sparse/linalg/isolve/lsmr.py:144]: If I remember this right, I added it as for some reason PyCharm was complaining about it. But for other files not.

**Review by rgommers — COMMENTED** [review]
The request for using the same tolerances between solvers seems sensible, but this is a small backwards compatibility break - existing code that managed to reach a `1e-6` tolerance can't necessarily reach `1e-8`. So this needs a release note at a minimum.

I'm not a user of these functions, would be good if someone else could have a look.

File: scipy/sparse/linalg/isolve/lsmr.py (hunk 1 above)
Comment (tupui) [inline scipy/sparse/linalg/isolve/lsmr.py:144]:
```suggestion
```

**Comment by tupui** [comment]
> The request for using the same tolerances between solvers seems sensible, but this is a small backwards compatibility break - existing code that managed to reach a `1e-6` tolerance can't necessarily reach `1e-8`. So this needs a release note at a minimum.

Do you mean `.. versionadded:: 1.7.0` in the docstring of the parameters?

**Comment by rgommers** [comment]
> Do you mean `.. versionadded:: 1.7.0` in the docstring of the parameters?

No, I mean an entry in the backwards incompatible changes section of https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.7.0 when we merge this PR.

**Comment by tupui** [comment]
@ilayn are you using this and can give a feedback here?

**Comment by tupui** [comment]
@perimosocordiae can you have a look?

**Comment by tupui** [comment]
Friendly reminder @ilayn @perimosocordiae

**Comment by tupui** [comment]
Any love for this one? 😅

**Comment by ilayn** [comment]
Oh my so sorry about this delay. I'll check asap.

**Comment by tupui** [comment]
> Oh my so sorry about this delay. I'll check asap.

No worries 😃 I am just going through my stash from time to time and got to this.

**Review by ilayn — APPROVED** [review]
I have two minor comments, 

1. probably it would be better if we relax both to 1e-6 rather than making lsmr stricter as some problems might stop returning a solution due to this out in the wild. A less approximate solution would hardly be a dramatic problem (famous last words).

2. probably we need that explanation of atol and btol in lsqr too in the docs.

Other than that, it looks good to me.

**Comment by ilayn** [comment]
The expalantion of the computation of atol * norm(a) ... + btol * norm(b) etc. is given in lsmr docs nicely but missing in lsqr (or I am  blind). I was wondering whether we should copy that into lsqr.

**Comment by tupui** [comment]
Better now @ilayn?

**Comment by ilayn** [comment]
Reads great. Let's wait a bit and then it would be ready to go

**Comment by tupui** [comment]
> Reads great. Let's wait a bit and then it would be ready to go

Great, thanks! I will let you merge when you want. And then I will add a note in the version notes.

**Comment by ilayn** [comment]
Let's see who is going to complain first, fingers crossed. Thanks @tupui !

**Comment by tupui** [comment]
Thanks for the review @ilayn

**Pull request merged**